kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms-test-support/src/main/java/io/kroxylicious/kms/provider/azure/kms/AzureKeyVaultKmsTestKmsFacade.java

```diff
@@ -0,0 +1,233 @@
+import com.sun.net.httpserver.HttpServer;
+
+import io.kroxylicious.kms.provider.azure.AzureKeyVaultEdek;
+import io.kroxylicious.kms.provider.azure.AzureKeyVaultKmsService;
+import io.kroxylicious.kms.provider.azure.WrappingKey;
+import io.kroxylicious.kms.provider.azure.config.AzureKeyVaultConfig;
+import io.kroxylicious.kms.provider.azure.config.auth.EntraIdentityConfig;
+import io.kroxylicious.kms.service.TestKekManager;
+import io.kroxylicious.kms.service.TestKmsFacade;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.proxy.config.secret.InlinePassword;
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultContainerBuilder.lowkeyVault;
+
+public class AzureKeyVaultKmsTestKmsFacade implements TestKmsFacade<AzureKeyVaultConfig, WrappingKey, AzureKeyVaultEdek> {
+
+    public static final Tls INSECURE_TLS = new Tls(null, new InsecureTls(true), null, null);
+    public static final String MOCK_AUTH_RESPONSE = """
+            {
+                "access_token": "aaa",
+                "token_type": "Bearer",
+                "expires_in": 3599,
+                "scope": "https%3A%2F%2Fgraph.microsoft.com%2Fmail.read",
+                "refresh_token": "bbb",
+                "id_token": "ccc"
+            }
+            """;
+    @Nullable
+    private LowkeyVaultContainer kms;
+    @Nullable
+    private HttpServer entraMock;
+
+    protected AzureKeyVaultKmsTestKmsFacade() {
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return DockerClientFactory.instance().isDockerAvailable();
+    }
+
+    public void startKms() {
+        this.kms = startVault();
+        try {
+            this.entraMock = HttpServer.create();
+            entraMock.createContext("/", exchange -> {
+                exchange.getResponseHeaders().add("Content-type", "application/json");
+                exchange.getRequestBody().readAllBytes();
+                exchange.getRequestBody().close();
+                byte[] responseBodyBytes = MOCK_AUTH_RESPONSE.getBytes(StandardCharsets.UTF_8);
+                exchange.sendResponseHeaders(200, responseBodyBytes.length);
+                exchange.getResponseBody().write(responseBodyBytes);
+                exchange.getResponseBody().close();
+            });
+            entraMock.bind(new InetSocketAddress(Inet4Address.getLocalHost(), 0), 1000);
+            entraMock.start();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public void stopKms() {
+        if (kms != null) {
+            kms.stop();
+        }
+    }
+
+    public LowkeyVaultContainer startVault() {
+        final DockerImageName imageName = DockerImageName.parse("nagyesta/lowkey-vault:4.0.0");
```

Comment:
Is this available in the google mirror?
We should add a renovate rule for this container image too.

Reply:
looks like it is:

```
podman pull mirror.gcr.io/nagyesta/lowkey-vault:4.0.0
Trying to pull mirror.gcr.io/nagyesta/lowkey-vault:4.0.0...
..
b2d9fda7634f9b9b2e6437739a139a50f555e85738d31253d948b5ef45d0f309
```

Reply:
makes me wonder if we need a kroxylicious-testcontainers module to hold extensions like the Image name substitutor. The integration suite is already pulling from the mirror, but the other modules aren't. I've made them directly reference the mirror for now.